src/PooledArrays.jl

```diff
@@ -373,6 +373,15 @@ function Base.push!(pv::PooledVector{S,R}, v::T) where {S,R,T}
     return v
 end
 
+function Base.append!(pv::PooledVector, items::AbstractArray)
+    itemindices = eachindex(items)
+    l = length(pv)
+    n = length(itemindices)
+    resize!(pv.refs, l+n)
```

Comment:
This would ensure entries are set to `#undef` in case conversion fails for some values:
```suggestion
    resize!(pv, l+n)
```
That's not what happens for `Array`s of `isbits` types, but maybe that's worth it here given that `PooledArray` is mostly useful for non-`isbits` types? Otherwise, leaving refs undefined could trigger crashes.

Reply:
I understand that you are afraid that later `copyto!` will not set the proper values there?

Reply:
Yes, if conversion fails it will stop in the middle AFAIK.